### PR TITLE
[Improvement-16870][RPC] Support retry configuration in rpc module

### LINKED ISSUE #16870: [Improvement][RPC] Support retry configuration in rpc module
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

We have some rpc method need to config retry times, e.g. retry times, retry interval.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request

close #16870 

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

## Pull Request Notice
[Pull Request Notice](https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/contribute/join/pull-request.md)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by CapitanJonhSilver — APPROVED** [review]

File: dolphinscheduler-extract/dolphinscheduler-extract-base/src/main/java/org/apache/dolphinscheduler/extract/base/RpcMethodRetryStrategy.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface RpcMethodRetryStrategy {
+
+    /**
+     * The maximum number of retries. Default is 3, which means that the method is retried at most 3 times, including the first call.
+     */
+    int maxRetryTimes() default 3;
```
Comment (Gallardot) [inline dolphinscheduler-extract/dolphinscheduler-extract-base/src/main/java/org/apache/dolphinscheduler/extract/base/RpcMethodRetryStrategy.java:35]: This means that by default all RPC methods will be retried 3 time. Are there currently any RPC methods that cannot be retried?

Comment (ruanwenjun) [reply dolphinscheduler-extract/dolphinscheduler-extract-base/src/main/java/org/apache/dolphinscheduler/extract/base/RpcMethodRetryStrategy.java:35]: Default only retry for `ConnectException`, this exception happens when connect to server failed.
image

So the retry is safa.

**Review by Gallardot — APPROVED** [review]
LGTM

**Review by SbloodyS — APPROVED** [review]
+1

**Pull request merged**